src/main/java/com/royalrangers/service/UserService.java

```diff
@@ -163,16 +164,15 @@ private UserAgeGroup determineUserAgeGroup(int userAge) {
         return rank;
     }
 
-    private List<User> getUsersToApproveByPlatoonID(Long platoonId) {
-        return userRepository.findAllByConfirmedTrueAndApprovedFalseAndPlatoonId(platoonId);
-    }
-
-    public List<User> getUsersForApprove(Long platoonId) {
-        return getUsersToApproveByPlatoonID(platoonId);
-    }
-
-    public List<User> getUsersForApproveForSuperAdmin() {
-        return userRepository.findAllByConfirmedTrueAndApprovedFalse();
+    public List<User> getUsersForApprove() {
+        User user = userRepository.findOne(getAuthenticatedUserId());
+        Set <Authority> roles = user.getAuthorities();
+        if (roles.contains(ROLE_SUPER_ADMIN))
+            return userRepository.findAllByConfirmedTrueAndApprovedFalse();
+        else {
```

Comment:
Lets check in else statement if **roles.contains(ROLE_ADMIN)**. It will be more readable and will avoid situations when someone call this method for normal user.